persistence/genesis.go

```diff
@@ -150,8 +144,22 @@ func (m *persistenceModule) populateGenesisState(state modules.PersistenceGenesi
 		log.Fatalf("an error occurred initializing flags: %s", err.Error())
 	}
 
-	if err = rwContext.Commit([]byte("TODO(#284): Genesis QC")); err != nil {
-		log.Fatalf("an error occurred during commit() on genesis state %s ", err.Error())
+	// Updates all the merkle trees
+	appHash, err := rwContext.ComputeAppHash()
+	if err != nil {
+		log.Fatalf("an error occurred updating the app hash during genesis: %s", err.Error())
+	}
+
+	// TODO_IN_THIS_COMMIT: Figure out what these values for genesis should be
+	genesisQuorumCert := []byte("placeholderQuorumCert")
+	genesisProposer := []byte("placeholderProposer")
```

Comment:
DISCUSS:

Do these affect the appHash at genesis?

Reply:
**Update:** I wasn't really happy with this answer so just wanted to provide an update that I'm looking deeper into this a bit.

Current thinking (not a proposal yet) is:

```
StateHash = what we have right now (the ordered concatination of all the state tries)
PrevAppHash = Hash(concat(StateHash, ProposerId, Hash(QC))
```

I still need more time to think through the different attacks that can take place though

---

In Cosmos, they have a variety of hahses, all mostly related to state:

<img width="740" alt="Screen Shot 2022-11-14 at 11 06 19 AM" src="https://user-images.githubusercontent.com/1892194/201746173-60960f67-506c-45bb-a80a-21c45085f45a.png">

In flow (hostuff in Go), they only have a payload hash but seem to be missing the links to the previous proposer and quorum cert:

<img width="544" alt="Screen Shot 2022-11-14 at 11 08 41 AM" src="https://user-images.githubusercontent.com/1892194/201746388-92e0217f-04a1-4b52-b8ba-cc3c53c6cc4d.png">

In Aptos (a hotstuff algorithm but in rust), they have a lot of hashes mainly related to the VM execution output:

<img width="439" alt="Screen Shot 2022-11-14 at 11 20 39 AM" src="https://user-images.githubusercontent.com/1892194/201747161-089f9342-0b8b-4cc9-98c4-c85d16e33c1f.png">